### ISSUE TITLE
When particles out of box, neighborlist builds failed

### ISSUE THREAD

**Issue by Roy-Kid** [google/jax-md#171]
Hi. I use JAX-MD to build a neighbor list, and find some particles that have no neighbors:
(we have 2988 particles in PDB file)
```
DeviceArray([2988, 2988, 2988, 2988, 2988, 2988, 2988, 2988, 2988, 2988,
             2988, 2988, 2988, 2988, 2988, 2988, 2988, 2988, 2988, 2988,
```
And I check those particles in ovito and they are all slightly out of the box. The box is 31.298 and one is 31.496. Then I use packmol to generate a box with padding to ensure all the particles in the box, then the result is correct. So I wonder that before using neighborlist_fn, how should I wrap all the particles back to the box by using JAX-MD. Or, I should carefully model the input. Thanks in advance!

**Comment by sschoenholz** [comment]
Hey Roy,

Thanks for your message. This actually just came up in #168. When using cell lists, the neighbor list code assumes that all points are inside the box. I think there are two options. First you could call `nbrs.update(jnp.mod(position, box_size))` to map particles into the box before constructing the neighbor list. Second, if you aren't using periodic boundary conditions you could use a larger box than you need to ensure that no particles are outside. 

Finally, if you are using periodic boundary conditions then using the `shift_fn` to move particles should ensure that everything stays within the simulation box.

Let me know if this helps or if you have any other questions.

**Comment by Roy-Kid** [comment]
Thank you so much for answering my question! We have fixed the bug and trying to a more elegant way to deal with this. If we make it, we will inform you ASAP!
